beacon_node/beacon_chain/src/data_availability_checker/overflow_lru_cache.rs

```diff
@@ -362,6 +361,7 @@ impl<E: EthSpec> PendingComponents<E> {
         )))
     }
 
+    /// Mark reconstruction as started for this `PendingComponent`.
```

Comment:
With this approach reconstruction will only be attempted once. I think that's okay, but we should carefully track in networks with metrics or crit logs if reconstruction errors. Worth it to add a comment explaining this

Reply:
Yep this was the intention as the reconstruction code would currently only fail due to code errors or kzg errors, which probably shouldn't be retried.

Good point adding this to the comment! I'll add an error log and metrics to track the failures. Not sure if it should be CRIT here though. (something that would trigger an alert to people monitoring infrastructure)